README.md

```diff
@@ -321,7 +321,22 @@ For information on the transition triggers, refer to:
 * description of the `timeout`, `ttl timeout`, and `ttr timeout` options in
 the sections of the corresponding [queue types](#queue-types).
 
-![Task state diagram](./doc/images/statediagram.svg)
+```mermaid
+flowchart LR
+      I(("init"))-->  |"put()"| R[READY]
+      I(("init"))--> |"put('my_task_data',{ delay = delay })"| DE[DELAYED]
+      R[READY]--> |"take()"| T[TAKEN]
+      R[READY]--> |"delete() / ttl timeout"| DO[DONE]
+      R[READY]--> |"bury()"| B[BURIED]
+      T[TAKEN]--> |"release() / ttr timeout"| R[READY]
+      T[TAKEN]--> |"release\n(id, { delay = delay })"| DE[DELAYED]
+      T[TAKEN]--> |"ack() / delete()"| DO[DONE]
+      T[TAKEN]--> |"bury()"| B[BURIED]
+      B[BURIED]--> |"delete() /\nttl timeout"| DO[DONE]
+      B[BURIED]--> |"kick()"| R[READY]
+      DE[DELAYED]--> |"timeout"| R[READY]
+      DE[DELAYED]--> |"delete()"| DO[DONE]
+```
```

Comment:
Isn't the state diagram fit better here?

```
stateDiagram-v2
      direction LR
      [*]-->READY: put()
      [*]-->DELAYED: put('my_task_data', {delay = delay})
      READY-->TAKEN: take()
      READY-->[*]: delete() / ttl timeout
      READY-->BURIED: bury()
      TAKEN-->READY: release() / ttr timeout
      TAKEN-->DELAYED: release(id, {delay = delay})
      TAKEN-->[*]: ack() / delete()
      TAKEN-->BURIED: bury()
      BURIED-->[*]: delete() / ttl timeout
      BURIED-->READY: kick()
      DELAYED-->READY: timeout
      DELAYED-->[*]: delete()
```

Rendered:

```mermaid
stateDiagram-v2
      direction LR
      [*]-->READY: put()
      [*]-->DELAYED: put('my_task_data', {delay = delay})
      READY-->TAKEN: take()
      READY-->[*]: delete() / ttl timeout
      READY-->BURIED: bury()
      TAKEN-->READY: release() / ttr timeout
      TAKEN-->DELAYED: release(id, {delay = delay})
      TAKEN-->[*]: ack() / delete()
      TAKEN-->BURIED: bury()
      BURIED-->[*]: delete() / ttl timeout
      BURIED-->READY: kick()
      DELAYED-->READY: timeout
      DELAYED-->[*]: delete()
```

Reply:
I don't like the crossed out words (captions) in this version.

![image](https://user-images.githubusercontent.com/57662928/167947291-2dfdb53e-f45f-4596-8e58-6c837900816b.png)

`DONE` is state, what for turn it into a point?

Reply:
Render the proposed version:
Rendered:

```mermaid
stateDiagram-v2
      direction LR
      [*]-->READY: put()
      [*]-->DELAYED: put('my_task_data', {delay = delay})
      READY-->TAKEN: take()
      READY-->[*]: delete() / ttl timeout
      READY-->BURIED: bury()
      TAKEN-->READY: release() / ttr timeout
      TAKEN-->DELAYED: release(id, {delay = delay})
      TAKEN-->[*]: ack() / delete()
      TAKEN-->BURIED: bury()
      BURIED-->[*]: delete() / ttl timeout
      BURIED-->READY: kick()
      DELAYED-->READY: timeout
      DELAYED-->[*]: delete()
```

Reply:
> I don't like the crossed out words (captions) in this version.

I initially looked into this, because squares around arrow notes feel as a bit too accented for me (I'm looking on it in dark GitHub theme). Anyway, it is a thing of taste. Nevermind.

I can only say that `stateDiagram-v2` variant looks visually easier in sources. However it is not enough to insist on this variant.

> `DONE` is state, what for turn it into a point?

Okay. I didn't know, whether it should be just final state and should have explicit name.

Reply:
(I'll resolve this conversation, because it is mostly about feelings, so should be up to the author.)

Reply:
In my case I went from `stateDiagram-v2` to `flowchart` and I found the second one more pleasant, let's ask @0x501D  opinion.

Yes,I use a dark theme too,but looked how it looks on a white background in the [editor](https://mermaid-js.github.io/mermaid-live-editor/edit#pako:eNqNkkFLxDAQhf9KyGVX2SJ4DChU2oO4eKh6WKwsYzOroU27pBOhlP53027TslXBXBryvvfyhqblWSWRC14TEEYKPgzo4Os6LdmwpDKYkapKtk382evlWxDcJnEY7QQ7WlpfnCtRvA13cXTSVrrZE9T5XgLBasNaiQU07IYN326yDnHO_Bw-xI-CEeS4_iG6fNEbkZzIrhhRwUhprCwt0buX5L7v8G5NMwcN6XN546Kg9llmmeXpaSDPK_n3IN40dIUsH8J95yX0e8vT6f_mndhxpFz1V3p1bD7LC_esn13FN1yj0aCkexltD6ecPlFjyoXbSjyALSjladk51B7dv8VYKqoMFwcoatxwsFQ9NWXGBRmLHhof2Eh13xB4umo).

Reply:
My first variant:
```
stateDiagram-v2
direction LR
r: READY
de: DELAYED
t: TAKEN
b: BURIED
do: DONE
[*] --> r: put()
[*] --> de: put('my_task_data',\n{ delay = delay })
r --> t: take()
r --> do: delete() /\nttl timeout
r --> b: bury()
t --> r: release() /\nttr timeout
t --> de: release\n(id, { delay = delay })
t --> do: ack() / delete()
t --> b: bury()
b --> do: delete() /\nttl timeout
b --> r: kick()
de --> r: timeout
de --> do: delete()
```

Reply:
Render:
```mermaid
stateDiagram-v2
direction LR
r: READY
de: DELAYED
t: TAKEN
b: BURIED
do: DONE
[*] --> r: put()
[*] --> de: put('my_task_data',\n{ delay = delay })
r --> t: take()
r --> do: delete() /\nttl timeout
r --> b: bury()
t --> r: release() /\nttr timeout
t --> de: release\n(id, { delay = delay })
t --> do: ack() / delete()
t --> b: bury()
b --> do: delete() /\nttl timeout
b --> r: kick()
de --> r: timeout
de --> do: delete()
```

Reply:
> 

To my mind  flowchart looks better.